runbotics/runbotics-orchestrator-ui/src/src-app/views/tenants/TenantsListView/TenantPluginsDrawer/TenantPluginsView/TenantPluginsView.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { VFC } from 'react';
+
+import AddIcon from '@mui/icons-material/Add';
+import CloseIcon from '@mui/icons-material/Close';
+import {
+    Button,
+    CircularProgress,
+    Divider,
+    IconButton,
+    Stack,
+    Typography,
+} from '@mui/material';
+
+import { addDays, isAfter, isBefore, isSameDay, parseISO } from 'date-fns';
+import { Tenant } from 'runbotics-common/dist/model/api/tenant.model';
+
+import useTranslations from '#src-app/hooks/useTranslations';
+import { useSelector } from '#src-app/store';
+import { tenantsSelector } from '#src-app/store/slices/Tenants';
+import {
+    PluginActivateButton, PluginBadge, PluginDate, PluginDrawerTitle,
+    PluginExpDate,
+    PluginsDrawer,
+} from '#src-app/views/tenants/TenantsListView/TenantPluginsDrawer/TenantPluginsView/TenantPluginsView.styles';
+
+interface TenantPluginsViewProps {
+    open: boolean,
+    onClose: () => void,
+    openEditDrawer: (data?) => void,
+    tenantData: Tenant;
+}
+
+const TenantPluginsView: VFC<TenantPluginsViewProps> = ({
+    open,
+    onClose,
+    openEditDrawer,
+    tenantData
+}) => {
+    const { tenantPlugins } = useSelector(tenantsSelector);
+    const { translate } = useTranslations();
+    const isExpired = (expDateStr?: string): boolean => {
+        if (!expDateStr) return false;
+        const expDate = parseISO(expDateStr);
+        const today = new Date();
+        return isBefore(expDate, today) && !isSameDay(expDate, today);
+    };
+
+    const isExpiringSoon = (expDateStr?: string): boolean => {
+        if (!expDateStr) return false;
+        const expDate = parseISO(expDateStr);
+        const today = new Date();
+        const inTwoWeeks = addDays(today, 14);
```

Comment:
what if we decide to change expiring soon date in future? We will need to change: variable name, its usage later in the code, and number. We should move this to constable before component function and name accordingly (meaning: without duration in variable name)